client/httplib/filter/log/filter.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020 beego
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httputil"
+
+	"github.com/beego/beego/v2/client/httplib"
+)
+
+// FilterChainBuilder is used to register logFilter to FilterChain
+type FilterChainBuilder struct {
+	PrintableContentTypes  []string               // only print the included request and response content types
+	MaxPrintableBodyLength int                    // print all content when value <= 0
+	Log                    func(v ...interface{}) // custom log function
+	Out                    io.Writer              // destination for output, only take effect when the Log function is nil
```

Comment:
only keep one filed.
The better way is that you define a NewBuilder function, it looks like
```go
func NewFilterBuilder(opts ...Opt) *FilterChainBuilder {
    res = newDefaultBuilder() // in this method, you will use all default value to build the instance.
    for _, o := range opts {
       o(res)
    }
}
```
Using the `Option` pattern. And then you could make FilterChainBuilder as private(optional).
The default value of `Log` field, it should be `logs.Debug`

Reply:
Considering that other filter builders also use `FilterBuilder`(not private), I think this should be unified.  Using option pattern will indeed be more in line with the Open/Close Principle.